pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -61,21 +70,31 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	return nil
 }
 
-// blank assignment to verify that ReconcileUserSignup implements reconcile.Reconciler
 var _ reconcile.Reconciler = &ReconcileUserSignup{}
 
+func NewUserSignupError(msg string) UserSignupError {
+	return UserSignupError{message: msg}
+}
+
+type UserSignupError struct {
```

Comment:
Don't start the name with the package name

Reply:
btw, what is the purpose of `UserSignupError`? What additional value does it bring?

Reply:
Renamed to just `SignupError`.  We return this error when reconciliation fails due to lack of member cluster availability.

Reply:
I meant what is the purpose of having a special type of error than returning a normal `error`?

Reply:
What is a "normal" error?  `error` is an interface, what error type do you think should be returned here?

Reply:
ah, I misunderstood the code. sorry - I thought that you wrapped another error, but you pass just a custom message. Sorry for the confusion and improper explanation.
I would probably use an `errorString` created by `fmt.Errorf(...)` as the error is only returned from the reconciler and there is no other processing of the error, but if you want to have a special type of the error (for any reason) then I'm fine... I just wanted to know why.